doc/developers/advanced_installation.rst

```diff
@@ -28,7 +28,7 @@ Installing a nightly build is the quickest way to:
 
 .. prompt:: bash $
 
-  pip install --pre --extra-index https://pypi.anaconda.org/scipy-wheels-nightly/simple scikit-learn
+  pip install --force-reinstall --no-deps --pre --extra-index https://pypi.anaconda.org/scipy-wheels-nightly/simple scikit-learn
```

Comment:
I am uneasy with this change. In a clean environment I get:

```python
> pip install --force-reinstall --no-deps --pre --extra-index https://pypi.anaconda.org/scipy-wheels-nightly/simple scikit-learn
Looking in indexes: https://pypi.org/simple, https://pypi.anaconda.org/scipy-wheels-nightly/simple
Collecting scikit-learn
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scikit-learn/1.2.dev0/scikit_learn-1.2.dev0-cp310-cp310-macosx_12_0_arm64.whl (17.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.0/17.0 MB 13.6 MB/s eta 0:00:00
Installing collected packages: scikit-learn
Successfully installed scikit-learn-1.2.dev0
> python
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:38:29) [Clang 13.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sklearn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/glemaitre/mambaforge/envs/xxx/lib/python3.10/site-packages/sklearn/__init__.py", line 82, in <module>
    from .base import clone
  File "/Users/glemaitre/mambaforge/envs/xxx/lib/python3.10/site-packages/sklearn/base.py", line 13, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```

IMHO, we should provide the minimum command to install the package. The state of the user environment should be managed by the users themselves. Here, we are touching to advance installation and there will always be a thousand ways to have it wrong :)

Reply:
Should we go back to the state proposed in 2ac3d43?

Reply:
I think this is a good proposal. More explicit than `--upgrade`, should work every time and experts can look at the commands and decide that they can do something different (at their own risk).

Reply:
Yes, moreover I think the commands at this revision are working in most situations and are idempotent.

Reply:
I'd add some hint about the fact that the command as it is in `main` won't do what you naively think it will do. This is based on my experience where it took me a while to realise why the command looked like it did the "right thing", but I was still seeing the bug that should have been fixed in the nightlies.

The pip output does tell you, but I didn't read it. If you have done something like this before you should know that this will happen. And so on. However, I'd claim that most people who read this bit of the docs have never used nightlies like this before or do it so rarely that they have forgotten what they learned last time.

This makes me think having `pip uninstall ... ; pip install ...` in the docs is a good move. Assuming we can't think of a case where this leads to broken things (like `--upgrade` does).

Reply:
> I am still unsure if we should change anything

> I'd add some hint about the fact that the command as it is in main won't do what you naively think it will do

Maybe we can leave the command as is but add a comment saying that uninstalling scikit-learn before running the command might be necessary to make sure to fetch to actually fetch the latest nightly build ?

Reply:
THe "add comment" option works for me

Reply:
I like adding a comment, because scikit-learn could have been installed in many ways: `conda`, `pip`, `apt`, etc, which requires different commands to uninstall.

Reply:
Fine with an additional comment.